internal/store/store_test.go

```diff
@@ -79,4 +84,46 @@ func TestStore(t *testing.T) {
 	if !errors.Is(err, types.ErrNotFound) {
 		t.Fatal("unexpected error", err)
 	}
+
+	t.Run("query all", func(t *testing.T) {
+		s, objs := createStoreWithRandomObjects(cdc, db, t, 50, "queryall")
+
+		results, err := s.Query(nil, 0, 0)
+		if err != nil {
+			t.Fatal("unexpected error", err)
+		}
+
+		i := 0
+		for ; results.Valid(); results.Next() {
+			if i == len(objs) {
+				t.Fatalf("Length mismatch, exepected %v elements but got more", len(objs))
+			}
+			var actual types.Object
+			if err := results.Read(actual); err != nil {
+				t.Fatal("Unexpected error :", err)
+			}
+			if !reflect.DeepEqual(actual, objs[i]) {
+				t.Fatalf("Object mismatch at index %v : expected = %v, actual = %v", i, objs[i], actual)
+			}
+			i++
+		}
+
+	})
+}
+
+func createStoreWithRandomObjects(cdc codec.Marshaler, db sdk.KVStore, t *testing.T, n int, uniqueID string) (Store, []types.Object) {
```

Comment:
@LeCodeurDuDimanche can this function be put into a common place so that it isn't duplicated across tests?

Reply:
I cannot make this function generic because `objects.Store` and `store.Store` are two different types that do not implement any common interface. 
I however factorized the object creation code to minimize the repetition between those two functions